motionplan/rrtStarConnect.go

```diff
@@ -224,12 +224,12 @@ func (mp *rrtStarConnectMotionPlanner) extend(algOpts *rrtStarConnectOptions, tr
 
 		// check to see if a shortcut is possible, and rewire the node if it is
 		_, connectionCost := algOpts.planOpts.DistanceFunc(&ConstraintInput{
-			StartInput: neighbor.node.q,
-			EndInput:   targetNode.q,
+			StartInput: neighbor.node.Q(),
+			EndInput:   targetNode.Q(),
 		})
 		cost := connectionCost + targetNode.cost
-		if cost < neighbor.node.cost && mp.checkPath(algOpts.planOpts, target, neighbor.node.q) {
-			neighbor.node.cost = cost
+		if cost < neighbor.node.(*costNode).cost && mp.checkPath(algOpts.planOpts, target, neighbor.node.Q()) {
```

Comment:
Just some general feedback, the pattern of accessing node, then accessing costnode, then accessing cost seems like it could be improved.